### PR TITLE
CB-11151 - Enhance the logging in the volume handler logic

### PR DESCRIPTION
**Pull request by gregito**
Enhance the logging in the volume handler logic with some additional refactors that removes code duplications and a NPE weak code snippet

### CODE REVIEW

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java [hunk 1]
```diff
@@ -183,17 +185,25 @@ public CloudResource delete(GcpContext context, AuthenticatedContext auth, Cloud
         for (VolumeSetAttributes.Volume volume : volumeSetAttributes.getVolumes()) {
             Future<Void> submit = intermediateBuilderExecutor.submit(() -> {
                 try {
+                    LOGGER.debug("Going to delete volume [id: {}] from project [id: {}] in the following availability zone: {}", volume.getId(),
+                            context.getProjectId(), volumeSetAttributes.getAvailabilityZone());
                     Operation operation = context.getCompute().disks()
                             .delete(context.getProjectId(), volumeSetAttributes.getAvailabilityZone(), volume.getId()).execute();
                     syncedOperations.add(operation.getName());
                     if (operation.getHttpErrorStatusCode() != null) {
+                        String message = String.format("%s [code: %d, message: %s, error: ]", VOLUME_DELETION_FAILED_BASE_MSG,
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java:194]: one string placeholder is missing

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java [hunk 2]
```diff
@@ -259,4 +269,9 @@ private Disk createDisk(String projectId, VolumeSetAttributes.Volume volume, Map
         }
         return result;
     }
+
+    private void logVolumeDeletionProblem(Exception e) {
+        LOGGER.info(String.format("%s, %s", VOLUME_DELETION_FAILED_BASE_MSG, e.getMessage()), e);
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java:274]: remove the comma from the format str

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java [hunk 3]
```diff
@@ -25,10 +30,25 @@ protected CloudResource createNamedResource(ResourceType type, String name) {
     }
 
     protected void exceptionHandler(GoogleJsonResponseException ex, String name, ResourceType resourceType) {
-        if (ex.getDetails().get("code").equals(HttpStatus.SC_NOT_FOUND)) {
-            LOGGER.debug("Resource {} not found: {}", resourceType, name);
+        throwIfNull(ex, () -> new IllegalArgumentException("Unable to handle exception due to: " + GoogleJsonResponseException.class.getSimpleName()
+                + " should not be null!"));
+        GoogleJsonError jsonError = ex.getDetails();
+        LOGGER.debug(GoogleJsonResponseException.class.getSimpleName() + " contains the following " + GoogleJsonError.class.getSimpleName() + ": " +
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java:36]: String.format ?

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java [hunk 4]
```diff
@@ -25,10 +30,25 @@ protected CloudResource createNamedResource(ResourceType type, String name) {
...
+                (jsonError != null ? jsonError.toString() : "null"));
+        if (jsonError != null) {
+            if (jsonError.get(HTTP_CODE_KEY) != null && jsonError.get(HTTP_CODE_KEY).equals(HttpStatus.SC_NOT_FOUND)) {
+                LOGGER.debug("Resource {} not found: {}", resourceType, name);
+            } else {
+                LOGGER.debug("Unable to uncover the error code of " + GoogleJsonResponseException.class.getSimpleName() + "! "
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java:42]: String.format?

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java [hunk 5]
```diff
@@ -25,10 +30,25 @@ protected CloudResource createNamedResource(ResourceType type, String name) {
...
+                        + GoogleJsonError.class.getSimpleName() + ": " + jsonError != null ? jsonError.toString() : "null");
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java:43]: jsonError is already not null here

File: cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceCheckerTest.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    @ParameterizedTest
+    @EnumSource(ResourceType.class)
+    void testExceptionHandlerWhenGoogleJsonErrorIsNullThenGcpResourceExceptionShouldCome(ResourceType resourceType) {
+        String expectedExceptionMessage = "somethingAwful";
+        when(mockGoogleJsonResponseException.getMessage()).thenReturn(expectedExceptionMessage);
+        GcpResourceException expectedException = assertThrows(GcpResourceException.class,
+                () -> underTest.exceptionHandler(mockGoogleJsonResponseException, RESOURCE_NAME, resourceType));
+
+        assertNotNull(expectedException.getMessage());
+        assertTrue(expectedException.getMessage().contains(expectedExceptionMessage));
```
Comment (bergerdenes) [inline cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceCheckerTest.java:68]: consider using AssertJ, it provides more usable assert fail messages (and has substring checks as well)

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java (hunk 1 above)
Comment (bergerdenes) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java:194]: added

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java (hunk 2 above)
Comment (bergerdenes) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpAttachedDiskResourceBuilder.java:274]: removed

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java (hunk 5 above)
Comment (bergerdenes) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java:43]: simplified

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java (hunk 4 above)
Comment (bergerdenes) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java:42]: applied

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java (hunk 3 above)
Comment (bergerdenes) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java:36]: formatted

File: cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceCheckerTest.java (hunk 6 above)
Comment (bergerdenes) [reply cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceCheckerTest.java:68]: used AssertJ

**Review by biharitomi — APPROVED** [review]
LGTM

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**